### PR TITLE
Disallow event signin/rsvp for pending+complete events for non-officers

### PR DESCRIPTION
**Pull request by thaigillespie**

### CODE REVIEW

File: src/pages/EventDetailsPage/components/EventDetails/index.tsx [hunk 1]
```diff
@@ -47,6 +52,30 @@ function EventDetailsComponent(props: EventDetailsComponentProps) {
     canva: { url: canvaURL, label: 'Canva' },
   };
 
+  const renderSignInButton = () => {
+    if (!isOfficer(userContext)) {
```
Comment (godwinpang) [inline src/pages/EventDetailsPage/components/EventDetails/index.tsx:56]: I think we can just disable both signin and rsvp as long as the status isn't ready

**Review by godwinpang — CHANGES_REQUESTED** [review]

File: src/pages/EventDetailsPage/components/EventDetails/index.tsx (hunk 1 above)
Comment (thaigillespie) [reply src/pages/EventDetailsPage/components/EventDetails/index.tsx:56]: So whether the user is officer or not doesn't matter? Okay

File: src/pages/EventDetailsPage/components/EventDetails/index.tsx [hunk 2]
```diff
@@ -72,12 +84,8 @@ function EventDetailsComponent(props: EventDetailsComponentProps) {
 
                     <Grid item>
                       <Grid container justify='flex-end' spacing={1}>
-                        <Grid item>
-                          <SignInButton eventId={eventId} />
-                        </Grid>
-                        <Grid item>
-                          <RSVPButton eventId={eventId} />
-                        </Grid>
+                        <Grid item>{renderSignInButton()}</Grid>
```
Comment (godwinpang) [inline src/pages/EventDetailsPage/components/EventDetails/index.tsx:87]: probably don't need to have 2 extra functions here, just add a `status === EventStatusEnum.Ready` ? <Grid ...> : <></>

**Review by godwinpang — CHANGES_REQUESTED** [review]

**Comment by thaigillespie** [comment]
Did it a bit differently than what you had, but there is only 1 function now @godwinpang

**Review by godwinpang — APPROVED** [review]

**Pull request merged**